### DISCUSSION TITLE
Dev update - internal server error

### DISCUSSION THREAD

**Discussion by yazend** [emsesp/EMS-ESP32#1013] [General]
I treyd upgrade to new dev Version, get an internal Server error. Don't know how to fix.
![Screenshot_20221002-213558_Samsung Internet](https://user-images.githubusercontent.com/5957256/193473122-8c62c04b-c89c-4b61-938b-0246f16262a7.jpg)

If logfiles needed let me know.
Alex

**Comment by MichaelDvP** [comment]
See #629

**Comment by yazend** [comment]
MichaelDvP thanks for the information,
I bought the Wifi only about 1 or 1½ months ago, instructions on how to flash it via USB would be helpful, I don't want to open the device, before I have a safe way to update it to the current DEV version.

**Comment by MichaelDvP** [comment]
You have to flash via usb. This is described [here](https://emsesp.github.io/docs/#/Uploading-firmware?id=via-a-usb-cable) and [here](https://bbqkees-electronics.nl/wiki/gateway/firmware-update-and-downgrade.html#updating-or-downgrading-the-gateway-firmware). **But don't use the Tasmota ESP-Flash-Tool!** It uses a wrong partition size, works for main, but not for dev build. A modified tool can be found [here](https://github.com/MichaelDvP/EMS-ESP_Flasher/releases/tag/Flash_size), Download the tool and the *.bin files to same directory.

**Comment by bbqkees** [comment]
@MichaelDvP I updated my wiki just now so it points to your tool.
@yazend no worries the update is pretty easy. Just unscrew the lid from the Gateway and connect the internal USB port to a pc.
There is nothing you can damage, in case a firmware update goes wrong you can still try again or load an older version of the firmware.

**Comment by proddy** [comment]
@MichaelDvP do you want to upload your EMS-ESP Flasher to a new repo in https://github.com/emsesp ?

**Comment by MichaelDvP** [comment]
Set here a version that loads the partitions online this repo? No need for for local files.

**Comment by yazend** [comment]
I will try it later on... let you guys know result. 
Many thanks for your quick support.
- FYI, you guys do a realy good job - thanks

**Comment by yazend** [comment]
OK, I got it working 
but new DEV version still have this issue.
![Screenshot 2022-10-03 192610](https://user-images.githubusercontent.com/5957256/193640814-27f42c68-82a4-4e52-ab62-f45b27c32e28.png)

can I follow this instuction and use new EMS-ESP-3_5_0b4-ESP32.bin as well?

![upgrade 001](https://user-images.githubusercontent.com/5957256/193641507-7c40e4ed-4074-4dd8-91d2-41d7becb60e2.JPG)

**Comment by proddy** [comment]
Yes

**Comment by MichaelDvP** [comment]
@yazend 
- If you have flashed the v3.5 once via usb, you can web-upload or OTA any version of ems-esp. 
- If you have the new bbQKees gateway with soldered esp32 you can use `partitions.esp32.16MB.bin`  in usb flashing,

**Comment by yazend** [comment]
Hey guys, many many thanks for your support, I am realy preciate.
